PathFinder.cc

```diff
@@ -11,38 +11,47 @@ namespace PathFinder {
 
 std::vector<PxRigidActor *> actors;
 
-float voxelHeight = 1.5;
-float voxelHeightHalf = voxelHeight / 2;
-float heightTolerance = 0.6;
-Vec start;
-Vec dest;
-float detectStep = 0.1;
-unsigned int iterDetect = 0;
-unsigned int maxIterDetect = 2000;
+Vec localVectorGetPath;
+Vec localVectorGenerateVoxelMap;
+Vec localVectorInterpoWaypointResult;
+Vec localVectorInterpoWaypointResult2;
+Vec localVectorDetect;
+Vec localVectorStep;
+Vec up = Vec(0, 1, 0);
+Matrix localmatrix;
+
+float heightTolerance;
+unsigned int maxIterStep;
+unsigned int *ignorePhysicsIds;
 unsigned int iterStep = 0;
-unsigned int maxIterStep = 1000;
 bool allowNearest = true;
-unsigned int maxVoxelCacheLen = 10000;
-bool isFound = false;
+unsigned int iterDetect = 0;
+unsigned int maxIterDetect;
+Vec start;
+Vec dest;
+Vec startGlobal;
+Vec destGlobal;
+float voxelHeightHalf;
 std::vector<Voxel *> frontiers;
 std::vector<Voxel *> voxels;
-std::vector<Voxel *> waypointResult;
 std::map<std::string, Voxel *> voxelo;
-Voxel localVoxel;
+Quat startDestQuaternion;
+bool isWalk = true;
+bool isFound = false;
+std::vector<Voxel *> waypointResult;
 Voxel *startVoxel;
 Voxel *destVoxel;
 PxBoxGeometry geom;
 unsigned int numIgnorePhysicsIds;
-unsigned int *ignorePhysicsIds;
+bool canLeft, canRight, canBtm, canTop, canBack, canFront;
 
-Vec localVector;
-Vec localVector2;
-
-void init(std::vector<PxRigidActor *> _actors, float _hy, float _heightTolerance, unsigned int _detectStep, unsigned int _maxIterdetect, unsigned int _maxIterStep, unsigned int _maxVoxelCacheLen, unsigned int _numIgnorePhysicsIds, unsigned int *_ignorePhysicsIds) {
+void init(std::vector<PxRigidActor *> _actors, float _hy, float _heightTolerance, unsigned int _maxIterdetect, unsigned int _maxIterStep, unsigned int _numIgnorePhysicsIds, unsigned int *_ignorePhysicsIds) {

```

Comment:
Instead of `init` it can be a constructor.

Reply:
Ok, I've tried make it a `class` instead of `namespace`, but encountered some problems.
I'll try again.

Reply:
Made `PathFinder` a `class` instead of `namespace`, and use `constructor` instead of `init()`.

https://github.com/webaverse/app-wasm/pull/35/commits/cbebe24928a5f0462a50b189dd7ab0c4e4a0f1c8
https://github.com/webaverse/app-wasm/pull/35/commits/acc2fbdfd11c87ae6d4cbea9e289b0c4f1145165
https://github.com/webaverse/app-wasm/pull/35/commits/fb232735da28b2e96501e4f92abb8b1568c57e6f